packages/fast-usdc/src/types.ts

```diff
@@ -0,0 +1,38 @@
+import type { ChainAddress } from '@agoric/orchestration';
+import type { IBCChannelID } from '@agoric/vats';
+import type { PendingTxStatus } from './constants.js';
+
+export type EvmHash = `0x${string}`;
+export type NobleAddress = `noble1${string}`;
+
+export interface CctpTxEvidence {
+  /** from Noble RPC */
+  aux: {
+    forwardingChannel: IBCChannelID;
+    recipientAddress: ChainAddress['value'];
+  };
+  blockHash: EvmHash;
+  blockNumber: bigint;
+  blockTimestamp: bigint;
+  chainId: number;
+  /** data covered by signature (aka txHash) */
+  tx: {
+    amount: bigint;
+    forwardingAddress: NobleAddress;
+  };
+  txHash: EvmHash;
+}
+
+export type LogFn = (...args: unknown[]) => void;
+
+export interface PendingTx extends CctpTxEvidence {
+  status: PendingTxStatus;
+}
+
+/** composite of NobleAddress and transaction amount value */
+export type PendingTxKey = `"["${NobleAddress}",${bigint}]"`;
```

Comment:
should not be parsable.
```suggestion
/** composite of NobleAddress and transaction amount value */
export type PendingTxKey = `"pendingTx:${string}`;
```